### PR TITLE
ClusterLoader - Test suite

### PR DESCRIPTION
**Pull request by krzysied**
ref #641

### CODE REVIEW

File: clusterloader2/cmd/clusterloader.go [hunk 1]
```diff
@@ -51,6 +51,7 @@ const (
 var (
 	clusterLoaderConfig config.ClusterLoaderConfig
 	testConfigPaths     []string
+	testSuiteConfigPath string
```
Comment (wojtek-t) [inline clusterloader2/cmd/clusterloader.go]: Add a todo to remove testconfigpaths after we finish migration.

Same for flags.

File: clusterloader2/cmd/clusterloader.go [hunk 2]
```diff
@@ -85,13 +86,17 @@ func initFlags() {
...
 	flags.StringArrayVar(&clusterLoaderConfig.TestOverridesPath, "testoverrides", []string{}, "Paths to the config overrides file. The latter overrides take precedence over changes in former files.")
+	flags.StringVar(&testSuiteConfigPath, "testsuite", "", "Path to the test suite config file")
 	initClusterFlags()
 }
 
 func validateFlags() *errors.ErrorList {
 	errList := errors.NewErrorList()
-	if len(testConfigPaths) < 1 {
-		errList.Append(fmt.Errorf("no test config path specified"))
+	if len(testConfigPaths) < 1 && testSuiteConfigPath == "" {
```
Comment (wojtek-t) [inline clusterloader2/cmd/clusterloader.go]: == 0 instead of < 1 :)

File: clusterloader2/cmd/clusterloader.go [hunk 3]
```diff
@@ -251,25 +256,20 @@ func main() {
...
-		} else {
-			specSummary.State = ginkgotypes.SpecStatePassed
-			printTestResult(clusterLoaderConfig.TestConfigPath, "Success", "")
+		for i := range testSuite {
+			clusterLoaderConfig.TestOverridesPath = testSuite[i].OverridePaths
+			clusterLoaderConfig.TestConfigPath = testSuite[i].ConfigPath
+			runSingleTest(f, prometheusFramework, junitReporter, suiteSummary)
+		}
+	} else {
+		for _, clusterLoaderConfig.TestConfigPath = range testConfigPaths {
```
Comment (wojtek-t) [inline clusterloader2/cmd/clusterloader.go]: this is super hacky  - can you make this assignment explicit

File: clusterloader2/pkg/config/template_provider.go [hunk 4]
```diff
@@ -161,6 +161,19 @@ func (tp *TemplateProvider) TemplateInto(path string, mapping map[string]interfa
 	return decodeInto(b, obj)
 }
 
+// GetTestSuite creates test suite config from file specified by the given path.
+func GetTestSuite(path string) (api.TestSuite, error) {
```
Comment (wojtek-t) [inline clusterloader2/pkg/config/template_provider.go]: ReadTestSuite ?

or maybe LoadTestSuite is better

File: clusterloader2/cmd/clusterloader.go (hunk 1 above)
Comment (krzysied) [reply clusterloader2/cmd/clusterloader.go]: Done.

File: clusterloader2/cmd/clusterloader.go (hunk 2 above)
Comment (krzysied) [reply clusterloader2/cmd/clusterloader.go]: Done.

File: clusterloader2/cmd/clusterloader.go (hunk 3 above)
Comment (krzysied) [reply clusterloader2/cmd/clusterloader.go]: Done.

File: clusterloader2/pkg/config/template_provider.go (hunk 4 above)
Comment (krzysied) [reply clusterloader2/pkg/config/template_provider.go]: Done + GetOverridesMapping -> LoadOverridesMapping to keep naming coherent.

File: clusterloader2/pkg/config/template_provider.go [hunk 5]
```diff
@@ -161,8 +161,21 @@ func (tp *TemplateProvider) TemplateInto(path string, mapping map[string]interfa
...
+	}
+	var testSuite api.TestSuite
+	if err = decodeInto(bin, &testSuite); err != nil {
+		return nil, err
+	}
+	return testSuite, nil
+}
+
+// LoadOverridesMapping returns mapping from file specified by the given paths.
+func LoadOverridesMapping(paths []string) (map[string]interface{}, error) {
```
Comment (wojtek-t) [inline clusterloader2/pkg/config/template_provider.go]: LoadTestOverrides  maybe?

[the word mapping doesn't appear anywhere in the api, flags, etc.]

Comment (krzysied) [reply clusterloader2/pkg/config/template_provider.go]: Done.

**Pull request merged**